src/hotspot/share/adlc/archDesc.cpp

```diff
@@ -24,6 +24,7 @@
 
 
 // archDesc.cpp - Internal format for architecture definition
+#include <unordered_set>
```

Comment:
Even though it's not hotspot code per se (but a tool used during build), it'll be the first usage of C++ STL in the hotspot code base.

HotSpot Coding Style [1] clearly states "Avoid using the C++ Standard Library." 

@kimbarrett what are your thought on it?

[1] https://github.com/openjdk/jdk/blob/master/doc/hotspot-style.md

Reply:
I think adlc is almost a stand alone tool, it only depends on src/hotspot/share/opto/opcodes.cpp . Could we move it to src/utils?

Reply:
I'm thinking if it can be out of hotspot and use stl in code.

Reply:
The rule is applied to general JDK sources and not just VM.
Please, check "C++ Standard Library" in `hotspot-style` Vladimir pointed and see if listed issues there applied to your changes.

Reply:
Ok, I will add a hashset implementation.

Reply:
Please wait. We are still discussing with other developers if it is OK to use STL in adlc.

Reply:
@kuaiwei  our general consensus is that you can use STL here since it is separate from VM tool.
@iwanowww do you approve?

Reply:
@iwanowww Can you check if I can use STL here?

Reply:
Nobody spoke up against, so I'm fine with using it.

Reply:
Thanks